src/plumpy/processes.py

```diff
@@ -1151,7 +1215,9 @@ def kill(self, msg: Union[str, None] = None) -> Union[bool, asyncio.Future]:
             self._state.interrupt(interrupt_exception)
             return cast(futures.CancellableAction, self._interrupt_action)
 
-        self.transition_to(process_states.ProcessState.KILLED, msg)
+        state_class = self.get_states_map()[process_states.ProcessState.KILLED]
+        new_state = self._create_state_instance(state_class, msg=msg)
+        self.transition_to(new_state)
```

Comment:
Compare to the one liner of original implementation, the new one seems more verbose. The state object needs to be created from the state tag with the parameters directly and passed to the `transition_to`. But, in my opinion, this API is a bit readable. The `transition_to` now only need to deal with the case where the passed in state is an object, instead of decide whether needs to create the state instance case by case.

Reply:
Yes, I did it by putting it into `_create_state_instance`, and it gets better. THanks.